src/vmm_mad/remotes/lib/vcenter_driver/virtual_machine.rb

```diff
@@ -128,6 +128,28 @@ class Nic < Resource
         def initialize(id, one_res, vc_res)
             super(id, one_res, vc_res)
         end
+
+        def self.one_nic(id, one_res)
+            @error_message = "vCenter device does not exist at the moment"
```

Comment:
would it be better to have a @nic_type as a class variable, instead of using @error_message?

Reply:
@error message is only an error to be displayed when u ask for a vcenter_device having only the one xml, thats why we only have @error_message in one_nic and one_disk.